### PR TITLE
Send fake bookmark event to client to reduce size of initEvents

### PR DESCRIPTION
**Pull request by Sindica**

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go [hunk 1]
```diff
@@ -1361,6 +1361,18 @@ func (c *cacheWatcher) process(ctx context.Context, initEvents []*watchCacheEven
 		klog.V(2).Infof("processing %d initEvents of %s took %v", len(initEvents), objType, processingTime)
 	}
 
+	// send fake bookmark event
```
Comment (yb01) [inline staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go:1364]: this is not fake anymore. it is a valid to keep the client's high watermark for idle watch sessions.

**Review by yb01 — APPROVED** [review]

**Review by yb01 — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go (hunk 1 above)
Comment (Sindica) [reply staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go:1364]: I still think it is fake as the event body was not supposed to be send to client. Unless we clean up the event body, it is not a real bookmark event.

**Pull request merged**